src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -196,6 +264,64 @@ protected boolean isVisible(Locator locator)
         }
     }
 
+    public EntityInsertPanel showGrid()
+    {
+        if (!isGridVisible())
+            elementCache().modeSelectListItem("from Grid")
+                    .waitForElement(this, 2000).click();
+        clearElementCache();
+        WebDriverWrapper.waitFor(()-> isGridVisible(),
+                "the grid did bot become visible", 2000);
+        return this;
+    }
+
+    public ResponsiveGrid uploadFileExpectingPreview(File file)
+    {
+        showFileUpload();
+        fileUploadPanel().uploadFile(file);
+        return new ResponsiveGrid.ResponsiveGridFinder(getDriver()).waitFor(this);
+    }
+
+    public EntityInsertPanel showFileUpload()
+    {
+        if (!isFileUploadVisible())
+        {
+            var toggle = elementCache().modeSelectListItem("from File")
+                    .waitForElement(this, 2000);
+            getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(toggle));
+            toggle.click();
+        }
+        clearElementCache();
+        WebDriverWrapper.waitFor(()-> isFileUploadVisible(),
+                "the file upload panel did bot become visible", 2000);
+        return this;
+    }
+
+    private void waitForLoaded()
+    {
+        WebDriverWrapper.waitFor(()-> {
+            try
+            {
+                return  isGridVisible() ||          // when uploading assay data there is no target select
+                        isFileUploadVisible() ||
+                        entityTypeSelect().isInteractive();
+            }catch (NoSuchElementException nse)
+            {
+                return false;
+            }
+        }, "The insert panel did not become loaded", WAIT_FOR_JAVASCRIPT);
+    }
+
+    /**
+     * finds the mode select tabs, to switch between grid input and file upload
+     * @param containsText
+     * @return
+     */
+    private Locator.XPathLocator modeSelectListItem(String containsText)
+    {
+        return Locator.tagWithClass("li", "list-group-item").containing(containsText);
+    }
```

Comment:
Why isn't this in the element cache?

Reply:
it is referenced by methods called from waitForReady.  Because that's called from the elementCache constructor, it doesn't work for it to be in the constructor